src/borrow_tracker/tree_borrows/mod.rs

```diff
@@ -285,42 +313,65 @@ trait EvalContextPrivExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
 
         let span = this.machine.current_span();
         let alloc_extra = this.get_alloc_extra(alloc_id)?;
-        let range = alloc_range(base_offset, ptr_size);
         let mut tree_borrows = alloc_extra.borrow_tracker_tb().borrow_mut();
 
-        // All reborrows incur a (possibly zero-sized) read access to the parent
-        if new_perm.initial_read {
+        let mut perform_parent_access = |range: AllocRange| -> InterpResult<'tcx> {
             tree_borrows.perform_access(
                 orig_tag,
                 Some((range, AccessKind::Read, diagnostics::AccessCause::Reborrow)),
                 this.machine.borrow_tracker.as_ref().unwrap(),
                 alloc_id,
                 this.machine.current_span(),
-            )?;
-        }
+            )
+        };
+
+        // Some reborrows incur a read access to the parent
```

Comment:
This doesn't just perform the read access, this does the full initialization, right?

Reply:
What do you mean by "full initialization"?  At this point, the
node hasn't been inserted into the tree yet; this happens in the
`new_child` method, called later.